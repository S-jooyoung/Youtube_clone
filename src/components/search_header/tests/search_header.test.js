// import dependencies
import React from "react";

// import react-testing methods
import { render, screen } from "@testing-library/react";

// import custom-component
import SearchHeader from "../search_header";
import userEvent from "@testing-library/user-event";

// import snapshot-testing methods
import renderer from "react-test-renderer";

describe("SearchHeader", () => {
  it("renders", () => {
    const component = renderer
      .create(<SearchHeader onSearch={jest.fn()} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
  describe("Input Sumit", () => {
    let onSearch;
    let input;
    let button;

    beforeEach(() => {
      onSearch = jest.fn();
      render(<SearchHeader onSearch={onSearch} />);
      input = screen.getByPlaceholderText("Search...");
      button = screen.getByTestId("test-button");
    });

    it("calls onSearch when button clicked", () => {
      userEvent.type(input, "BTS");
      userEvent.click(button);

      expect(onSearch).toHaveBeenCalledWith("BTS");
    });
  });
});
