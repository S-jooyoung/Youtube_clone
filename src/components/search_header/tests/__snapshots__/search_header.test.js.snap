// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader renders 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <img
      alt="logo"
      className="img"
      src="/images/logo.png"
    />
    <h1
      className="title"
    >
      Youtube
    </h1>
  </div>
  <input
    className="input"
    onKeyPress={[Function]}
    placeholder="Search..."
    type="text"
  />
  <button
    className="button"
    data-testid="test-button"
    onClick={[Function]}
    type="submit"
  >
    <img
      alt=""
      className="buttonImg"
      src="/images/search.png"
    />
  </button>
</header>
`;
